contrib/lib/tests/templates.rs

```diff
@@ -116,5 +116,57 @@ mod templates_tests {
             let response = client.get("/tera/test").dispatch();
             assert_eq!(response.status(), Status::NotFound);
         }
+
+        #[cfg(debug_assertions)]
+        #[test]
+        fn test_template_reload() {
+            use rocket::local::Client;
+            use std::fs::File;
+            use std::io::Write;
+            use std::path::Path;
+            use std::thread;
+            use std::time::Duration;
+
+            fn write_file(path: &Path, text: &str) {
+                let mut file = File::create(path).expect("open file");
+                file.write_all(text.as_bytes()).expect("write file");
+                file.sync_all().expect("sync file");
+            }
+
+            const RELOAD_TEMPLATE: &'static str = "hbs/reload";
+            const INITIAL_TEXT: &'static str = "initial";
+            const NEW_TEXT: &'static str = "reload";
+
+            let reload_path = Path::join(
+                Path::new(env!("CARGO_MANIFEST_DIR")),
+                "tests/templates/hbs/reload.txt.hbs"
+            );
+
+            write_file(&reload_path, INITIAL_TEXT);
```

Comment:
Add a comment above this. I suppose it's `// set up the template with the initial text _before_ initializing the Rocket instance so that it's picked up on the initial load`.